docs/email-signup.md

```diff
@@ -0,0 +1,23 @@
+# Email signup
```

Comment:
I think this section would have a better home in email-alert-api, as even though the business finder is the only thing using them, the features aren't limited to just finders.

Reply:
Think it's useful encapsulating all the business finder specific stuff in finder-frontend rather than somewhere like email-alert-api 🤔 but happy to move individual sections to other repos where it makes sense.

For example, have now updated these docs to point to https://github.com/alphagov/email-alert-api/blob/master/doc/matching-content-to-subscriber-lists.md and have cut down the amount of email-alert-api explanation here.

Things like `EmailAlertListTitleBuilder` live in finder-frontend, so even though it's an 'emaily' feature probably should be documented in this repo & not email-alert-api.